### PR TITLE
AL-852: GWCS inverse transform should respect its bounding box

### PR DESCRIPTION
**Pull request by nden**
GWCS has a bug where the inverse transform does not respect the bounding_box. This leads to unexpected results that affect other applications. The GWCS bud is fixed in https://github.com/spacetelescope/gwcs/pull/498
This PR fixes problems in jwst which surfaced when the bug was fixed.

**Checklist for PR authors (skip items if you don't have permissions or they are not applicable)**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [x] updated or added relevant tests
- [ ] updated relevant documentation
- [x] added relevant milestone
- [x] added relevant label(s)
- [x] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] All comments are resolved
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

**Comment by emolter** [comment]
The failing oldestdeps tests appear to show that asdf needs a minimum of at least 3.3 now since that's what GWCS needs

**Review by emolter — COMMENTED** [review]
ASDF still needs a bump to >=3.3.0

It looks like your auto-formatter touched CHANGES.rst, which is making the CI unhappy

Needs a changelog fragment

I forgot to remove two imports that are no longer used in test_outlier_detection.py, namely `create_fitswcs()` and `compute_s_region_imaging()`

Other than those small things that are needed to make our CI pass, this looks good to me.

File: jwst/resample/resample_utils.py
Comment (emolter) [inline jwst/resample/resample_utils.py:168]: what would it take to make the relevant function accept the `with_bounding_box=False` argument?

File: jwst/tweakreg/tests/test_multichip_jwst.py
Comment (emolter) [inline jwst/tweakreg/tests/test_multichip_jwst.py:379]:
```suggestion
```

File: jwst/tweakreg/tests/test_multichip_jwst.py
Comment (emolter) [inline jwst/tweakreg/tests/test_multichip_jwst.py:463]:
```suggestion
```

File: jwst/resample/resample_utils.py
Comment (nden) [reply jwst/resample/resample_utils.py:168]: The question has been raised with other developers in the community and the answer is always that there shouldn't be kwargs passed to these fnuctions. Part of the reason is that the shared API should work with other WCS libraries and the bounding_box is gwcs specific.

**Comment by nden** [comment]
Regression test run: https://github.com/spacetelescope/RegressionTests/actions/runs/12395581275/job/34601785739

MIRI LRS differences are expected, since the bounding_box has changed.
NRS differences are: 1 px in one case and 4 px in the other. 
NRS moving target differences are bigger, worth looking into ths.

**Comment by emolter** [comment]
> NRS moving target differences are bigger, worth looking into ths.

Without doing too deep a dive, just plotting some of the outputs, it looks to me like as we guessed in stand-up there are just a few more flagged pixels in the data which are then propagating to the output spectra and it's not much to worry about.  But I'll let Melanie have the final word on this one

**Review by WilliamJamieson — APPROVED** [review]
These changes look good for me other than the comments made by @emolter which look relatively trivial.

Does this warrant a change log fragment?

File: jwst/resample/resample_utils.py
Comment (mcara) [inline jwst/resample/resample_utils.py:171]: I thought `deepcopy`ing a WCS is cpu-intense. Wouldn't it be better to you `try-finally` to turn it off/on? Still, I think @emolter 's comment about turning this off may be hiding some other issue is relevant.

File: jwst/resample/tests/test_resample_step.py
Comment (mcara) [inline jwst/resample/tests/test_resample_step.py:836]: This is way, way too large for a test. I think it is an indication that `crval` is wrong if you need to make output shape this large in order to get data within the output.

**Review by mcara — COMMENTED** [review]
I think most changes are reasonable except for the custom wcs test where output shape is increased to 10k pixels. I believe the original test does not work because of innacurate `crval` and that issue about the bbox calculation in `stcal`.

File: jwst/resample/resample_utils.py
Comment (mcara) [reply jwst/resample/resample_utils.py:168]: What harm would occur if shared API would have `with_bounding_box` but it would have no effect on, e.g., FITS WCS? Especially if the default is `with_bounding_box=False`. Then only users who explicitly need this feature will set `with_bounding_box=True` (with no effect on FITS WCS).

Comment (mcara) [reply jwst/resample/resample_utils.py:168]: What harm would occur if shared API would have with_bounding_box but it would have no effect on, e.g., FITS WCS? Especially if the default is with_bounding_box=False. Then only users who explicitly need this feature will set with_bounding_box=True (with no effect on FITS WCS).

**Review by mcara — APPROVED** [review]
I think `test_custom_wcs_resample_imaging` needs better parameters that do not require these large output shapes. I also think that we need to investigate why it is necessary to disable bounding box for pixmap computations. However, this should prevent merging this PR as it allows the pipeline to work as before the major GWCS change.

**Comment by emolter** [comment]
Attempting to summarize a conversation we have between @mcara, @braingram, @tapastro, @perrygreenfield and I this morning.

This PR fixes multiple issues.
We are happy with the changes that set `with_bounding_box=False`.

The other change is to unset the `bounding_box` before calling `wcs.world_to_pixel_values`.  This reproduces the old behavior of the code, so we think it's mostly ok to merge it for now.  However it basically bypasses/hides a bug where the pixmap is sometimes all NaNs.  This stems from the fact that we are not updating the `bounding_box` for nonzero values of `crpix`.

The latter bug is fixed by https://github.com/spacetelescope/stcal/pull/326, but it is currently causing some unexpected Romancal failures and there might be additional discussion needed before merging that one.

The proposed plan of action is to:

1. Ensure the latest regtest failures on this PR are indeed expected - we are a little bit suspicious of the MIRI LRS s_region differences, because they look large
2. Merge this PR basically as-is
3. Finish build 11.2 - all the other changes can wait
4. Finalize https://github.com/spacetelescope/stcal/pull/326, then see what happens if we revert the unsetting of `bounding_box` in this PR after that fix is merged
5. Decide what should happen with https://github.com/spacetelescope/stcal/pull/324, which also involves bypassing the `bounding_box` in various functions.  It is not clear to me whether this would be necessary if the other bug were fixed.

Also tagging @melanieclarke to see if that sounds good

File: pyproject.toml
Comment (melanieclarke) [reply pyproject.toml:27]: Thanks, I saw this was still failing, but was waiting to see what we wanted to do with this PR before trying anything else.  I will try numpy 1.24 now.

File: jwst/resample/tests/test_resample_step.py
Comment (melanieclarke) [reply jwst/resample/tests/test_resample_step.py:836]: I can take a look.  I have worked with these tests somewhat recently.

Comment (melanieclarke) [reply jwst/resample/tests/test_resample_step.py]: I pushed up a fix that keeps the custom parameters to more reasonable values and checks that the output data is nontrivial.

File: pyproject.toml
Comment (melanieclarke) [reply pyproject.toml]: Oldestdeps are building now.

**Review by emolter — APPROVED** [review]
I'm happy with this given that we think we know the cause of the LRS bounding_box differences

**Comment by melanieclarke** [comment]
I looked at the MIRI LRS differences in spec2 S_REGION, from this regression test run:
    https://github.com/spacetelescope/RegressionTests/actions/runs/12402085588

I agree that some changes are expected, since there are changes to bounding boxes and the bounding box matters a lot for MIRI LRS, but I don't know the MIRI WCS well enough to validate that the new version is correct.  I recommend that we go ahead and merge this, since it fixes other issues and impact to data is otherwise minor, and ask the MIRI team for help validating the S_REGION, either in testing for this build, or in collaboration on other planned changes for the next build.

@tapastro - opinions?

**Comment by tapastro** [comment]
I agree - we'll alert MIRI LRS of the differences for their testing, but they don't appear to be large enough to warrant further investigation before merge.

**Review by tapastro — APPROVED** [review]
LGTM

**Comment by nden** [comment]
The bounding box of the MIRI LRS fixed slit was slightly larger that the range of `points` in the TabularModel that defines the WCS. This causes the footprint to be all NaNs. So it's sensible to use the range of the TabularModel as the bounding box.

**Pull request merged**
